cloudify_azure/auth/oauth2.py

```diff
@@ -195,7 +200,116 @@ def request_access_token(self):
         return data
 
 
-class ServicePrincipalCertificateAuth(AADMixin):
+class CustomAADMixin(object):
```

Comment:
How does this differ from the standard AADMixin?

Reply:
This is in order to override both `_configure` and `_create_adal_context` methods to handle lookup the cloud_envrioment and `endpoints_active_directory`